bundles/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/handler/EvohomeHeatingZoneHandler.java

```diff
@@ -82,16 +89,115 @@ public void handleCommand(ChannelUID channelUID, Command command) {
+                                        getNextSetPointChange(schedule));
+                                break;
+                            case EvohomeBindingConstants.SETPOINT_OVERRIDE_TEMPORARY_TIME_ADD:
+                                bridge.setTemporarySetPoint(getEvohomeThingConfig().id, newTemp,
+                                        LocalDateTime.now().plusMinutes(getOverrideTime()));
+                                break;
+                            default:
+                                bridge.setPermanentSetPoint(getEvohomeThingConfig().id, newTemp);
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private DailySchedules getDailySchedules() {
+        return this.getEvohomeBridge().getZoneSchedule("temperatureZone", this.getId());
+    }
+
+    private LocalDateTime getNextSetPointChange() {
+        return getNextSetPointChange(this.getDailySchedules());
+    }
+
+    private LocalDateTime getNextSetPointChange(DailySchedules schedule) {
+        if (schedule != null) {
+            int currentWeekDay = LocalDateTime.now().getDayOfWeek().getValue() - 1;
+            LocalDateTime nextSetPointChange = null;
+            for (DailySchedule daily : schedule.getSchedules()) {
+                // handle wrap around
+                int dayDelta = 0;
+                if (daily.getWeekday() < currentWeekDay) {
+                    // the schedule day is from earlier in the week so add 7 to get to the same day next week
+                    dayDelta = daily.getWeekday() + 7 - currentWeekDay;
+                } else {
+                    dayDelta = daily.getWeekday() - currentWeekDay;
+                }
+                LocalDate setPointDay = LocalDate.now().plusDays(dayDelta);
+                for (Switchpoint sp : daily.getSwitchpoints()) {
+                    LocalDateTime spNextOccurance = LocalDateTime.of(setPointDay,
+                            LocalTime.parse(sp.getTimeOfDay().toString()));
+                    if (spNextOccurance.isAfter(LocalDateTime.now())) {
+                        // setPoint is after now we are interested in it.
+                        if (nextSetPointChange == null || nextSetPointChange.isAfter(spNextOccurance)) {
+                            nextSetPointChange = spNextOccurance;
+                        }
                     }
                 }
             }
+            if (nextSetPointChange == null) {
+                return LocalDateTime.now().plusMinutes(getOverrideTime());
+            }
+            return nextSetPointChange;
+        } else {
+            return LocalDateTime.now().plusMinutes(getOverrideTime());
+        }
+    }
+
+    private double getCurrentSetPoint() {
+        return getCurrentSetPoint(this.getDailySchedules());
+    }
+
+    private double getCurrentSetPoint(DailySchedules schedule) {
```

Comment:
For this method as weel, could this be moved to `DailySchedules`. It shares some overlap with the previous method in terms of code, so please move the shared code (such as the `dayDelta`) to methods.

Reply:
yes it could move to DailySchedules , however the code is not shared, although the structure is similar the search order is negated